### PR TITLE
Add docs for Scheme

### PR DESCRIPTION
**Pull request by aalekseyev**
Signed-off-by: Arseniy Alekseyev <aalekseyev@janestreet.com>

### CODE REVIEW

File: src/scheme.mli [hunk 1]
```diff
@@ -3,10 +3,28 @@
...
   | Approximation of Dir_set.t * 'rules t
+
+  (** [Finite rules] just produces a fixed set of rules known in advance. *)
   | Finite of 'rules Path.Build.Map.t
+
+  (** [Thunk f] is a "lazy" collection of rules. This is normally used with
+      [Approximation (dirs, Thunk f)] such that the work of [f] can be delayed
+      (or avoided entirely) until (or unless) the rules for [dirs] become
+      necessary. *)
   | Thunk of (unit -> 'rules t)
```
Comment (rgrinberg) [inline src/scheme.mli]: Is it guaranteed that the argument of `Thunk` will only be called once?

Comment (aalekseyev) [reply src/scheme.mli]: Once per `evaluate`, yes.

Comment (rgrinberg) [reply src/scheme.mli]: Okay, I think it's worth clarifying.

**Comment by rgrinberg** [comment]
Thanks for the docs

**Pull request merged**